### PR TITLE
Add boilerplate code for Dataflux-Pytorch Lightning demo

### PR DESCRIPTION
**Pull request by abhibyreddi**

### CODE REVIEW

File: demo/lightning/data.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import torch
+import lightning.pytorch as pl
+
+from dataflux_pytorch import dataflux_mapstyle_dataset
+
+
+class Unet3DDataModule(pl.LighitningDataModule):
+    def __init__(self, gcs_bucket):
+        self.data_dir = gcs_bucket + "/images"
```
Comment (bernardhan33) [inline demo/lightning/data.py:9]: Use os.path.join() for directory joins.

**Review by bernardhan33 — APPROVED** [review]

File: demo/lightning/data.py (hunk 1 above)
Comment (MattIrv) [inline demo/lightning/data.py:9]: Should this path be configurable, with this as the default?

File: demo/lightning/train.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+from model import Unet3D
+from data import Unet3DDataModule
+
+if __name__ == "__main__":
+    model = Unet3D()
+    train_data_loader = Unet3DDataModule()
+    trainer = pl.Trainer(
+        devices=2, 
+        accelerator="gpu", 
+        max_epochs=5
```
Comment (MattIrv) [inline demo/lightning/train.py:12]: I think these should all be configurable too

File: demo/lightning/data.py
Comment (MattIrv) [inline demo/lightning/data.py:1]: Regarding the file path, JD is working on a similar demo for text-based training now, so maybe this should live in its own sub-folder within demo/lightning?

**Review by MattIrv — APPROVED** [review]

**Review by abhibyreddi — COMMENTED** [review]
Moved the three new files from demo/lightning to demo/lightning/image-segmentation. Will probably have to move demo/image-segmentation to demo/pytorch/image-segmentation in a later PR.

I agree that the arguments to Unet3D and Unet3DDataModule constructors should be configurable. I will make that change after I have this workload up and running with hardcoded parameters.

**Pull request merged**